cmake/DPF-plugin.cmake

```diff
@@ -355,22 +404,30 @@ function(dpf__build_vst3 NAME DGL_LIBRARY)
   dpf__set_module_export_list("${NAME}-vst3" "vst3")
   target_link_libraries("${NAME}-vst3" PRIVATE "${NAME}-dsp" "${NAME}-ui")
   set_target_properties("${NAME}-vst3" PROPERTIES
-    LIBRARY_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/bin/$<0:>"
     ARCHIVE_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/obj/vst3/$<0:>"
-    OUTPUT_NAME "${NAME}-vst3"
+    OUTPUT_NAME "${NAME}"
     PREFIX "")
-  # TODO set correct output directory for VST3 packaging
-  #if(APPLE)
-    #set_target_properties("${NAME}-vst3" PROPERTIES
-      #LIBRARY_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/bin/${NAME}.vst3/Contents/MacOS/$<0:>"
-      #OUTPUT_NAME "${NAME}"
-      #SUFFIX "")
+
+  if(APPLE)
+    set_target_properties("${NAME}-vst3" PROPERTIES
+      LIBRARY_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/bin/${NAME}.vst3/Contents/MacOS/$<0:>"
+      SUFFIX "")
+  elseif(WIN32)
+    set_target_properties("${NAME}-vst3" PROPERTIES
+      LIBRARY_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/bin/${NAME}.vst3/Contents/${vst3_arch}-win/$<0:>")
+  else()
+    set_target_properties("${NAME}-vst3" PROPERTIES
+      LIBRARY_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/bin/${NAME}.vst3/Contents/${vst3_arch}-linux/$<0:>")
```

Comment:
hmm this is not quite correct due to DPF supporting basically anything posix compatible..
I know Steinberg has no spec outside of mac, windows and linux, but that perhaps should not stop us?
Or maybe we raise the issue upstream.

Reply:
Unless mistaken, it would appear that at least FreeBSD already supports the "linux" naming.
The port of sfizz has the VST build option, and we don't have a complaint by yuri about it.

Reply:
tbf yuri just packages things without much concern, so even without concerns it still doesnt seem right.
does bsd even have a vst3 host at this point? (not sure how ardour or qtractor work there)
if someone wants to ship a multi-os binary that includes bsd and linux at the same time, well that is just not possible if both are defined as "linux"

on bsd it would seem logical to just use the "bsd" string instead of "linux". but for a few other systems it is not so clear.
the vst3 spec just needs to say what format or style to use for not-yet-inside-vst3-defined-systems.
lets raise an issue upstream, I will do that now.

Reply:
opened ticket at https://github.com/steinbergmedia/vst3sdk/issues/86